metaseq/dataclass/configs.py

```diff
@@ -737,3 +748,49 @@ class MetaseqConfig(MetaseqDataclass):
     lr_scheduler: Any = MISSING
     bpe: Any = MISSING
     tokenizer: Any = None
+
+
+class DynamicConfig(dict):
+    """
+    can be used instead of redis to store updatable key-value
+
+    test:
+        data = {"a": 1}
+        with open('data.json', 'w') as fp:
+            json.dump(data, fp)
+        dcfg = DynoCfg(json_file_path = "data.json", timeout = 1)
+        print(dcfg["a"]) # > 1
+        data = {"a": 4}
+        with open('data.json', 'w') as fp:
+            json.dump(data, fp)
+        print(dcfg["a"]) # > 1
+        time.sleep(1)
+        print(dcfg["a"]) # > 4
+    """
+
+    default_state = {"force_profile": False}
+
+    def __init__(self, json_file_path=None, timeout=30):  # timeout in sec
+        self.timeout = timeout
+        self.timer_start = time.time()
+        self.json_file_path = json_file_path
+        if self.json_file_path is not None:
+            with open(self.json_file_path, "r") as json_file:
+                super().__init__(**json.load(json_file))
+        else:
+            super().__init__(**DynamicConfig.default_state)
+
+    def refresh(self):
+        if self.json_file_path is not None:
+            # if data have not been updated in a while, reload from the file
+            # can speed up if were to calculate hash sum of the file to monitor updates
+            timer_value = time.time() - self.timer_start
+            if timer_value > self.timeout:
+                with open(self.json_file_path, "r") as json_file:
+                    refreshed_config = json.load(json_file)
+                    super().__init__(**refreshed_config)
```

Comment:
Is it safe to re-init objects?

Reply:
Did not have problems with this when testing, but I think I will change this when issue the next patch